### PR TITLE
Team command update

### PR DESCRIPTION
**Pull request by davight**
## Changes:
- Update to `autoExecute`
- Remove prefix / suffix limit
- Change teams name character limit to check only on versions lower than 1.18

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java [hunk 1]
```diff
@@ -70,156 +72,90 @@ public TeamCommand() {
...
+            scoreboard = ScoreboardHelper.getMain();
         }
         else {
-            if (ScoreboardHelper.hasScoreboard(id.asString())) {
-                board = ScoreboardHelper.getScoreboard(id.asString());
-            }
-            else {
-                board = ScoreboardHelper.createScoreboard(id.asString());
-            }
+            scoreboard = ScoreboardHelper.hasScoreboard(id.asString()) ? ScoreboardHelper.getScoreboard(id.asString()) : ScoreboardHelper.createScoreboard(id.asString());
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: I'm not sure why you renamed and reformatted things here?

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java [hunk 2]
```diff
@@ -70,156 +72,90 @@ public TeamCommand() {
...
-        if ((prefix != null && prefix.length() > 64) || (suffix != null && suffix.length() > 64)) {
-            throw new InvalidArgumentsException("Prefixes and suffixes must be 64 characters or less!");
-        }
-        if (scriptEntry.hasObject("option") != scriptEntry.hasObject("status")) {
-            throw new InvalidArgumentsException("Option and Status arguments must go together!");
-        }
-        scriptEntry.defaultObject("id", new ElementTag("main"));
-    }
-
-    public static String translateEntry(String entry, TagContext context) {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java:146]: This method should've been updated, not removed - it was useful organization of repeated logic that's now just duplicated below

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java [hunk 3]
```diff
@@ -70,156 +72,90 @@ public TeamCommand() {
...
+            switch (option.asString().toLowerCase()) {
+                case "friendly_fire" -> {
+                    team.setAllowFriendlyFire(status.toString().equalsIgnoreCase("always"));
+                }
+                case "see_invisible" -> {
+                    team.setCanSeeFriendlyInvisibles(status.toString().equalsIgnoreCase("always"));
+                }
+                default -> {
+                    if (option.matchesEnum(Team.Option.class)) {
+                        team.setOption(Team.Option.valueOf(option.asString().toUpperCase()), status);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: asEnum

**Comment by davight** [comment]
fixed

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java [hunk 4]
```diff
@@ -229,10 +153,23 @@ else if (optName.equals("see_invisible")) {
...
+            return obj.asType(PlayerTag.class, context).getName();
+        }
+        else if (obj.shouldBeType(EntityTag.class)) {
+            return obj.asType(EntityTag.class, context).getUUID().toString();
+        }
+        else {
+            return obj.toString();
+        }
+    }
+
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: stray newline

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java [hunk 5]
```diff
@@ -193,33 +112,38 @@ public void execute(ScriptEntry scriptEntry) {
...
-            String statusName = status.asLowerString();
-            if (optName.equals("friendly_fire")) {
-                team.setAllowFriendlyFire(statusName.equals("always"));
-            }
-            else if (optName.equals("see_invisible")) {
-                team.setCanSeeFriendlyInvisibles(statusName.equals("always"));
-            }
-            else {
-                team.setOption(Team.Option.valueOf(optName.toUpperCase()), Team.OptionStatus.valueOf(statusName.toUpperCase()));
+            switch (option.asString().toLowerCase()) {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: `asLowerString`

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java [hunk 6]
```diff
@@ -193,33 +112,38 @@ public void execute(ScriptEntry scriptEntry) {
...
+                case "friendly_fire" -> {
+                    team.setAllowFriendlyFire(status.toString().equalsIgnoreCase("always"));
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: never use `equalsIgnoreCase`, use `asLowerString().equals`

Comment (davight) [reply plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: should I convert it to elementtag then?

Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: ... wait, that's an enum? use the enum checks then not strings.

**Comment by davight** [comment]
fixed

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java (hunk 4 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java]: you removed the wrong newline here

**Comment by davight** [comment]
sorry, fixed again

**Pull request merged**